examples/plot_complex_data.py

```diff
@@ -0,0 +1,191 @@
+# %%
```

Comment:
Examples should always start by a docstring gives a high level description of the example and make its education objective explicit.

For instance:
```suggestion
"""
==========================================
Modeling competing risks on synthetic data
==========================================

This example introduces a synthetic data generation tool that can be helpful to study
the relative performance and potential biases of predictive competing risks estimators
on right-censored data.

Some of the input features and their second order multiplicative interactions are
statistically associated with the parameters of the distributions from which the
competing events are sampled.

# %%
```

We can refine and expand more once we have converged on the content of the example.

Reply:
Accepting the above suggestion should fix the following failing CI job as visible in the pull request status panel:

![image](https://github.com/soda-inria/hazardous/assets/89061/816bc1f5-1438-42b3-ab2f-23c34cb7c39d)

The "details" of this line currently links to the following error message:

![image](https://github.com/soda-inria/hazardous/assets/89061/def1c9b9-e680-4b9f-9d85-f6e4bec261a0)

Once fixed, this CI job should generate the HTML rendered version of this example online which will help the discussion on the PR.